packages/mirinae/src/inputs/dropdown/select-dropdown/PSelectDropdown.vue

```diff
@@ -136,68 +140,145 @@ const handlePressDownKey = () => {
         }
     });
 };
+const handleDeleteTag = (index: number) => {
+    state.proxySelected = undefined;
+    state.selectedItems.splice(index, 1);
+};
+const handleDeleteButtonClick = () => {
+    state.proxySelected = undefined;
+    state.selectedItems = [];
+    emit('click-delete');
+};
+
+const hideMenu = () => {
+    state.proxyVisibleMenu = false;
+};
+onClickOutside(containerRef, hideMenu);
+
+/* Watcher */
+watch(() => state.proxySelected, (proxySelected) => {
+    if (proxySelected === undefined || null) return;
+
+    if (!props.items) {
+        state.selectedItems = [];
+        return;
+    }
+
+    let item;
+    if ((props.appearanceType === APPEARANCE_TYPE.BADGE) || (props.appearanceType === APPEARANCE_TYPE.STACK)) {
+        if (props.indexMode) {
+            item = props.items[proxySelected ?? 0];
+            return;
+        }
+        item = props.items.filter((i) => i.name === proxySelected)[0];
+
+        state.selectedItems.push(item);
+        return;
+    }
+
+    if (props.indexMode) {
+        item = props.items[proxySelected ?? 0];
+        state.selectedItems = [item];
+        return;
+    }
+
+    const data = groupBy(props.items, 'name')[proxySelected ?? ''];
+    if (Array.isArray(data)) {
+        state.selectedItems = data;
+    }
+}, { immediate: true });
```

Comment:
**Suggestion:** Upon inspecting the internals of this `watch` hook, it seems to detect changes in the `proxySelected` state and modifies the `selectedItems` accordingly. I've reviewed the code as a whole, and it appears that `selectedItems` is modified in the `handleDeleteTag` and `handleDeleteButtonClick` functions. However, within these functions, the state is also modified alongside `proxySelected`.

Given that both `proxySelected` and `selectedItems` appear to serve the same purpose of holding selected items, managing two states for the same purpose can lead to various issues. It can result in accidental updates to the wrong state depending on the situation.

In such cases, computed properties are commonly used. In this scenario, you might consider keeping `selectedItems` as the primary state and adding a computed property called `singleSelectItem`. This computed property could work by observing changes in `selectedItems` while allowing you to perform the necessary actions, such as splicing specific indexes when needed, as in the `handleDeleteTag` function.

This approach can help simplify your state management and reduce the risk of inadvertent updates to the wrong state.

**Original Korean Comment:** 이 `watch` 훅의 내부를 살펴보면, `proxySelected`의 상태를 감지하여 `selectedItems`의 값을 변경해주고 있습니다. 
어떤 경우에 `selectedItems`를 변경하고 있는지를 코드 전체적으로 살펴봤는데요.
`handleDeleteTag`, `handleDeleteButtonClick` 함수에서 변경하고 있습니다.
그런데 이 함수 내에서도 `proxySelected`와 함께 상태가 변경됩니다.
그렇다면 `proxySelected`와 `selectedItems`는 사실상 '선택된 항목을 담는 상태 값'이라는 같은 목적을 가지고 있는 것 같습니다.
같은 목적을 가지고 있는데 상태가 이중으로 관리되면 여러가지 문제를 야기합니다.
상황에 따라 필요한 상태만 변경하는 실수를 저지를 수도 있고요.
이런 경우에 주로 활용하는 것이 `computed` 입니다.

이 경우에는 `handleDeleteTag` 함수에서 특정 `index` 를 통해 `selectedItems` 항목을 `splice` 해줘야 하므로 별도 상태로 추가된게 아닐까 싶습니다.
이런 상황일 때는 `selectedItems` 를 원장으로 두고, `singleSelectItem` 이라는 `computed` 상태를 추가하여 `selectedItems`를 바라보면서 동작하도록 하면 어떨까 의견 드립니다.

Reply:
**Summary:** It would be beneficial to make `selectedItems` the primary state, with `selectedItem` (formerly `proxySelected`) added as a complementary state observing changes in `selectedItems`. Additionally, the events previously emitted by `proxySelected` can be handled through a watch hook observing changes in `selectedItem`.

**요약:** 요약하면 `selectedItems` 가 원장이 되도록 변경하고, `selectedItem`(기존의 `proxySelected`)이 추가되어 `selectedItems`를 바라보도록 하면 좋을 것 같습니다. 그리고 기존의 `proxySelected`가 emit하던 이벤트를 `selectedItem`을 바라보는 `watch` 훅으로 해결할 수 있을 것 같습니다.